pytorch_lightning/trainer/training_loop.py

```diff
@@ -504,6 +504,9 @@ def run_training_epoch(self):
             if batch_output.signal == -1:
                 break
 
+            # update non-plateau LR schedulers
+            self.update_lr_schedulers('step', skip_plateau_scheduler=True)
```

Comment:
Why here?

Should be right above `VALIDATE IF NEEDED` (L526)

Reply:
Because every_n_train_steps based checkpointing happens in on_train_batch_end, so we need this before on_train_batch_end.
Specifically, in the test added every_n_train_steps=2 case will fail if this was added after on_train_batch_end.